crates/matrix-sdk/src/room/timeline/event_item.rs

```diff
@@ -159,101 +158,140 @@ impl EventTimelineItem {
     /// Get the raw JSON representation of the initial event (the one that
     /// caused this timeline item to be created).
     ///
-    /// Returns `None` if this event hasn't been echoed back by the server yet.
+    /// Returns `None` if this event hasn't been echoed back by the server
+    /// yet.
     pub fn raw(&self) -> Option<&Raw<AnySyncTimelineEvent>> {
-        self.raw.as_ref()
+        match self {
+            Self::Local(local_event) => local_event.raw.as_ref(),
+            Self::Remote(remote_event) => remote_event.raw.as_ref(),
+        }
     }
 
-    pub(super) fn to_redacted(&self) -> Self {
-        Self {
-            // FIXME: Change when we support state events
-            content: TimelineItemContent::RedactedMessage,
-            reactions: BundledReactions::default(),
-            ..self.clone()
+    /// Clone the current event item, and update its `content`.
+    pub(super) fn with_content(&self, content: TimelineItemContent) -> Self {
+        match self {
+            Self::Local(local_event_item) => {
+                Self::Local(LocalEventTimelineItem { content, ..local_event_item.clone() })
+            }
+            Self::Remote(remote_event_item) => {
+                Self::Remote(RemoteEventTimelineItem { content, ..remote_event_item.clone() })
+            }
         }
     }
+}
 
-    pub(super) fn with_transaction_id_event_id(
-        &self,
-        txn_id: &OwnedTransactionId,
-        event_id: Option<OwnedEventId>,
-    ) -> Self {
-        Self {
-            key: TimelineKey::new_transaction_id_with_event_id(txn_id.clone(), event_id),
-            ..self.clone()
-        }
+#[derive(Clone)]
+pub struct LocalEventTimelineItem {
+    /// The transaction ID.
+    pub transaction_id: OwnedTransactionId,
+    /// The event ID received from the server in the event-sending response.
+    pub event_id: Option<OwnedEventId>,
+    /// The sender of the event.
+    pub sender: OwnedUserId,
+    /// The sender's profile of the event.
+    pub sender_profile: Profile,
+    /// The timestamp of the event.
+    pub timestamp: MilliSecondsSinceUnixEpoch,
+    /// The content of the event.
+    pub content: TimelineItemContent,
+    /// Encryption information.
+    pub encryption_info: Option<EncryptionInfo>,
```

Comment:
I believe we only have `EncryptionInfo` on remote events since we construct the local event before it is encrypted.

Reply:
Removed, thanks.